CMakeLists.txt

```diff
@@ -15,9 +15,19 @@
 cmake_minimum_required(VERSION 3.25)
 cmake_policy(SET CMP0079 NEW) # Allow target_link_libraries() in subdirs
 
+include(src/cmake/feature-flags.cmake)
 include(src/cmake/environment-utils.cmake)
 is_running_in_ci(IS_RUNNING_IN_CI)
 
+# A feature flag with a dependency.
+feature_flag(APPLEVZ_ENABLED "AppleVZ backend" APPLE)
+# A feature flag with no dependencies.
+feature_flag(AVAILABILITY_ZONES_ENABLED "Availability zone support")
+
+# Print the features' status to show this off.
+message(STATUS "APPLEVZ_ENABLED = ${APPLEVZ_ENABLED}")
```

Comment:
Is the intention to then litter:

```
#if(APPLEVZ_ENABLED)
  list(APPEND MULTIPASS_BACKENDS apple)
#endif
```

like changes when we need to make a change required by a feature flag?